packages/core/src/invoke.ts

```diff
@@ -151,20 +156,47 @@ export function spawnPromise<T>(
   };
 }
 
+export function spawnActivity<TC, TE extends EventObject>(
+  activityCreator: (ctx: TC, event: TE) => any
+): InvokeCreator<TC, TE> {
+  return (ctx, e, { parent, id }) => {
+    let dispose;
+    try {
+      dispose = activityCreator(ctx, e);
+    } catch (err) {
+      parent.send(error(id, err) as any);
+    }
+
+    return {
+      id,
+      send: () => void 0,
+      toJSON: () => ({ id }),
+      subscribe() {
+        // do nothing
+        return {
+          unsubscribe: () => void 0
+        };
+      },
+      stop: isFunction(dispose) ? () => dispose() : undefined
+    };
+  };
+}
+
 export function spawnCallback<TE extends EventObject = AnyEventObject>(
-  callback: (ctx: any, e: any) => InvokeCallback
+  callbackCreator: (ctx: any, e: any) => InvokeCallback
 ): InvokeCreator<any, AnyEventObject> {
-  return (_, __, { parent, id }) => {
+  return (ctx, event, { parent, id, _event }) => {
+    const callback = callbackCreator(ctx, event);
     let canceled = false;
     const receivers = new Set<(e: EventObject) => void>();
     const listeners = new Set<(e: EventObject) => void>();
 
-    const receive = (e: TE) => {
-      listeners.forEach(listener => listener(e));
+    const receive = (receivedEvent: TE) => {
+      listeners.forEach(listener => listener(receivedEvent));
       if (canceled) {
         return;
       }
-      parent.send(e);
+      parent.send(receivedEvent);
```

Comment:
i've looked at this really quickly, but it seems that this is the only actor type which doesn't add `origin` to the `receivedEvent`

Reply:
To be removed